### PR TITLE
Add new recipe for bamstats

### PR DESCRIPTION
**Pull request by emi80**
Add recipe for [bamstats](/guigolab/bamstats)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by emi80** [comment]
@BiocondaBot please add label

File: recipes/bamstats/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{% set name = "bamstats" %}
+{% set version = "0.3.5" %}
+
+package:
+  name: {{name}}
+  version: {{ version }}
+
+build:
+  number: 0
+  binary_relocation: false
```
Comment (bgruening) [inline recipes/bamstats/meta.yaml:10]: This one is not needed.

File: recipes/bamstats/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - {{name}} --help
+    - {{name}} --version
+
+about:
+  home: https://github.com/guigolab/bamstats
+  license: BSD-3-Clause
+  summary: A command line tool to compute mapping statistics from a BAM file
+
+extra:
+  skip-lints:
```
Comment (bgruening) [inline recipes/bamstats/meta.yaml:32]: Are those skips really needed?

File: recipes/bamstats/meta.yaml (hunk 1 above)
Comment (emi80) [reply recipes/bamstats/meta.yaml]: Removed `binary_relocation` option

File: recipes/bamstats/meta.yaml (hunk 2 above)
Comment (emi80) [reply recipes/bamstats/meta.yaml]: Removed extra section

File: recipes/bamstats/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+test:
+  commands:
+    - {{name}} --help
+    - {{name}} --version
+
+about:
+  home: https://github.com/guigolab/bamstats
+  license: BSD-3-Clause
+  summary: A command line tool to compute mapping statistics from a BAM file
```
Comment (cbrueffer) [inline recipes/bamstats/meta.yaml:28]:
```suggestion
  summary: A command line tool to compute mapping statistics from a BAM file
  license_file: LICENSE
```

File: recipes/bamstats/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{% set name = "bamstats" %}
+{% set version = "0.3.5" %}
+
+package:
+  name: {{name}}
```
Comment (cbrueffer) [inline recipes/bamstats/meta.yaml:5]:
```suggestion
  name: {{ name }}
```

File: recipes/bamstats/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  version: {{ version }}
+
+build:
+  number: 0
+
+source:
+  - url: https://github.com/guigolab/{{name}}/archive/v{{ version }}.tar.gz
```
Comment (cbrueffer) [inline recipes/bamstats/meta.yaml:12]:
```suggestion
  - url: https://github.com/guigolab/{{ name }}/archive/v{{ version }}.tar.gz
```

File: recipes/bamstats/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    sha256: 0b397c0b09c4a944e564dd9e92eb5e9e8408e97cf1a54af07dced8783342b977
+    folder: src/github.com/guigolab/{{name}}/
+
+requirements:
+  build:
+    - {{ compiler('go') }}
+
+test:
+  commands:
+    - {{name}} --help
```
Comment (cbrueffer) [inline recipes/bamstats/meta.yaml:22]:
```suggestion
    - bamstats --help
```
I'd suggest using the actual binary here to make things more explicit.

File: recipes/bamstats/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    - {{name}} --version
```
Comment (cbrueffer) [inline recipes/bamstats/meta.yaml:23]:
```suggestion
    - bamstats --version
```

File: recipes/bamstats/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  name: {{name}}
+  version: {{ version }}
+
+build:
+  number: 0
+
+source:
+  - url: https://github.com/guigolab/{{name}}/archive/v{{ version }}.tar.gz
+    sha256: 0b397c0b09c4a944e564dd9e92eb5e9e8408e97cf1a54af07dced8783342b977
+    folder: src/github.com/guigolab/{{name}}/
```
Comment (cbrueffer) [inline recipes/bamstats/meta.yaml:14]:
```suggestion
    folder: src/github.com/guigolab/{{ name }}/
```

**Review by cbrueffer — COMMENTED** [review]
Looks like a lot of comments, but it's mostly just style and adding the license :-)

**Comment by emi80** [comment]
HI @cbrueffer, thanks. I think I addressed all comments. It's true it seemed a lot of stuff but it was quick. 😄

File: recipes/bamstats/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+test:
+  commands:
+    - bamstats --help
+    - bamstats --version
+
+about:
+  home: https://github.com/guigolab/bamstats
+  license: BSD-3-Clause
+  summary: A command line tool to compute mapping statistics from a BAM file
+  license_file: LICENSE
```
[the remark this answers is not in the document]
Comment (emi80) [reply recipes/bamstats/meta.yaml:29]: Not really... I think I copied it from another Golang-based recipe.
I assume I can just remove the `folder` attribute from the `source` section, right?

[the remark this answers is not in the document]
Comment (cbrueffer) [reply recipes/bamstats/meta.yaml:29]: Yes, the paths in the build script would also need to be adapted.

[the remark this answers is not in the document]
Comment (emi80) [reply recipes/bamstats/meta.yaml:28]: It should work now

**Review by cbrueffer — APPROVED** [review]

**Comment by cbrueffer** [comment]
Looks good, thanks for the changes!

**Pull request merged**